src/common/vector_operations/vector_hash.cpp

```diff
@@ -176,6 +176,33 @@ static inline void ListLoopHash(Vector &input, Vector &hashes, const SelectionVe
 	}
 }
 
+template <bool HAS_RSEL, bool FIRST_HASH>
+static inline void ArrayLoopHash(Vector &input, Vector &hashes, const SelectionVector *rsel, idx_t count) {
+	auto hdata = FlatVector::GetData<hash_t>(hashes);
+
+	UnifiedVectorFormat idata;
+	input.ToUnifiedFormat(count, idata);
+
+	// Hash the children into a temporary
+	auto &child = ArrayVector::GetEntry(input);
+	auto array_size = ArrayType::GetSize(input.GetType());
+	auto child_count = array_size * count;
+
+	Vector child_hashes(LogicalType::HASH, child_count);
+	if (child_count > 0) {
+		VectorOperations::Hash(child, child_hashes, child_count);
```

Comment:
We probably need to add this here
```c++
child_hashes.Flatten(child_count);
```
I recently added this to `ListLoopHash` in #8979 to fix a bug with joining lists. The problem was that the child of the list became `CONSTANT` by the `VectorOperations::Hash`, but in the rest of the function it was assumed that it was `FLAT`.